eyeshade/controllers/referrals.js

```diff
@@ -1,23 +1,68 @@
 const boom = require('boom')
 const bson = require('bson')
+const BigNumber = require('bignumber.js')
 const Joi = require('@hapi/joi')
 const underscore = require('underscore')
+const _ = underscore
 
 const utils = require('bat-utils')
 const braveHapi = utils.extras.hapi
 const braveJoi = utils.extras.joi
+const extrasUtils = utils.extras.utils
+
+const queries = require('../lib/queries')
 
 const v1 = {}
 
-const schema = Joi.array().min(1).items(Joi.object().keys({
+const amountValidator = braveJoi.string().numeric()
+const groupNameValidator = Joi.string().optional().description('the name given to the group')
+const channelIdValidator = braveJoi.string().publisher().required().description('the publisher identity')
```

Comment:
I think `publisherIdValidator` might be a more opt name for this since we're validating the publisher identity and not a channel